src/Bridge/Laravel/DbToolsServiceProvider.php

```diff
@@ -0,0 +1,270 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MakinaCorpus\DbToolsBundle\Bridge\Laravel;
+
+use Illuminate\Contracts\Config\Repository;
+use Illuminate\Contracts\Foundation\Application;
+use Illuminate\Foundation\Console\AboutCommand;
+use Illuminate\Support\ServiceProvider;
+use MakinaCorpus\DbToolsBundle\Anonymization\AnonymizatorFactory;
+use MakinaCorpus\DbToolsBundle\Anonymization\Anonymizer\AnonymizerRegistry;
+use MakinaCorpus\DbToolsBundle\Anonymization\Config\Loader\ArrayLoader;
+use MakinaCorpus\DbToolsBundle\Anonymization\Config\Loader\PhpFileLoader;
+use MakinaCorpus\DbToolsBundle\Anonymization\Config\Loader\YamlLoader;
+use MakinaCorpus\DbToolsBundle\Bridge\Symfony\DependencyInjection\DbToolsConfiguration;
+use MakinaCorpus\DbToolsBundle\Command\Anonymization\AnonymizeCommand;
+use MakinaCorpus\DbToolsBundle\Command\Anonymization\AnonymizerListCommand;
+use MakinaCorpus\DbToolsBundle\Command\Anonymization\CleanCommand;
+use MakinaCorpus\DbToolsBundle\Command\Anonymization\ConfigDumpCommand;
+use MakinaCorpus\DbToolsBundle\Command\BackupCommand;
+use MakinaCorpus\DbToolsBundle\Command\CheckCommand;
+use MakinaCorpus\DbToolsBundle\Command\RestoreCommand;
+use MakinaCorpus\DbToolsBundle\Command\StatsCommand;
+use MakinaCorpus\DbToolsBundle\Configuration\Configuration;
+use MakinaCorpus\DbToolsBundle\Configuration\ConfigurationRegistry;
+use MakinaCorpus\DbToolsBundle\Database\DatabaseSessionRegistry;
+use MakinaCorpus\DbToolsBundle\Error\ConfigurationException;
+use MakinaCorpus\DbToolsBundle\Storage\DefaultFilenameStrategy;
+use MakinaCorpus\DbToolsBundle\Storage\FilenameStrategyInterface;
+use MakinaCorpus\DbToolsBundle\Storage\Storage;
+use Symfony\Component\Config\Definition\Processor;
+
+class DbToolsServiceProvider extends ServiceProvider
+{
+    public function register(): void
+    {
+        $this->app->extend('config', function (Repository $config, Application $app) {
+            $dbToolsConfig = $config->get('db-tools', []);
+            $dbToolsConfig = ['db_tools' => $dbToolsConfig];
+
+            $processor = new Processor();
+            $configuration = new DbToolsConfiguration(false, true);
+
+            $dbToolsConfig = $processor->processConfiguration($configuration, $dbToolsConfig);
+            $dbToolsConfig = DbToolsConfiguration::fixLegacyOptions($dbToolsConfig);
+            $dbToolsConfig = DbToolsConfiguration::appendPostConfig($dbToolsConfig);
```

Comment:
I was thinking about this, maybe the `fixLegacyOptions()` method should be implicitly called within the `appendPostConfig()` method?

Reply:
That seems relevant to me.

Reply:
yep, that seems relevant to me too

Reply:
@SimonMellerin I renamed `appendPostConfig()` method to `finalizeConfiguration()`, since it is now intended to apply all types of modification we could have to do after the configuration file processing.